packages/core/src/mesh/ModelMesh.ts

```diff
@@ -14,4 +65,800 @@ export class ModelMesh extends Mesh {
     super(engine);
     this.name = name;
   }
+
+  /**
+   * Upload Mesh Data to the graphics API.
+   * @param noLongerAccessible - Whether to access data later. If true, you'll never access data anymore (free memory cache)
+   */
+  uploadMeshData(noLongerAccessible: boolean): void {
+    const { _indices: _indicesArray, _vertexSlotChanged, _accessible } = this;
+    if (!_accessible) {
+      throw "Not allowed to access data while accessible is false.";
+    }
+
+    // Structure of vertex buffer has changed
+    if (_vertexSlotChanged || this._vertexCountChanged) {
+      const vertexElements = this._updateVertexElements();
+      const elementCount = this._elementCount;
+      let vertices = this._verticesFloat32;
+      if (!vertices || vertices.length !== elementCount * this._vertexCount) {
+        vertices = new Float32Array(elementCount * this._vertexCount);
+        this._verticesUint8 = new Uint8Array(vertices.buffer);
+        this._verticesFloat32 = vertices;
+      }
+      this._vertexChangeFlag = ValueChanged.All;
+      this._updateVertices(vertices);
+      this._vertexBuffer = new Buffer(
+        this._engine,
+        BufferBindFlag.VertexBuffer,
+        vertices,
+        noLongerAccessible ? BufferUsage.Static : BufferUsage.Dynamic
+      );
+      this.setVertexElements(vertexElements);
+      this.setVertexBufferBinding(this._vertexBuffer, elementCount * 4);
+      this._vertexSlotChanged = false;
+      this._vertexCountChanged = false;
+    } else {
+      const verticesArray = this._verticesFloat32;
+      this._updateVertices(verticesArray);
+      this._vertexBuffer!.setData(verticesArray);
```

Comment:
"this._vertexBuffer.setData(verticesArray);" can in  this._updateVertices method. Thus  you can easy use the vertexChangFlag to know if  need call setData.

Reply:
No need to do so.